src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -1375,4 +1394,119 @@ private ITextComponent onGetDisplayName4(IMerchant villager) {
         }
         return new TextComponentString(SpongeTexts.toLegacy(this.displayName));
     }
+
+    @Override
+    @Nullable
+    public ProfileProperty getSkin() {
+        return Iterables.getFirst(this.getProfile().getPropertyMap().get(ProfileProperty.TEXTURES), null);
+    }
+
+    @Override
+    public boolean setSkin(ProfileProperty skin) {
+        Collection<ProfileProperty> props = this.getProfile().getPropertyMap().get(ProfileProperty.TEXTURES);
+        props.clear();
+        props.add(skin);
+
+        this.updateSkin();
+        return true;
+    }
+
+    @Override
+    public DataTransactionResult removeSkin() {
+        if (!this.getProfile().getPropertyMap().containsKey(ProfileProperty.TEXTURES)) {
+            return DataTransactionResult.successNoData();
+        }
+        Collection<ProfileProperty> skin = this.getProfile().getPropertyMap().removeAll("textures");
+        ImmutableValue<?> oldValue = new ImmutableSpongeValue<>(Keys.SKIN, skin.iterator().next());
+        this.updateSkin();
+
+        return DataTransactionResult.builder().result(DataTransactionResult.Type.SUCCESS).replace(oldValue).build();
+    }
+
+    private void updateSkin() {
+        this.updateSkinSelf();
+        this.updateSkinOthers();
+    }
+
+    private void updateSkinSelf() {
+        this.getTabList().removeEntry(this.getUniqueId());
+        this.getTabList().addEntry(TabListEntry.builder()
+                .displayName(this.getDisplayNameData().displayName().get())
```

Comment:
Does this will work with custom plugins?
I think, old entry values should be used.

Reply:
And what if self entry was removed by some plugin?

Reply:
Good point - I'll push a fix taking that into account.